### DISCUSSION TITLE
How can i include a citation inside a caption of a table generated by "kable"?

### DISCUSSION THREAD

**Discussion by piiskop** [quarto-dev/quarto-cli#957] [Q&A]
Sometimes, it's necessary to cite the source where a table's data come from. And usually, i use _kable_ for creating tables. In _bookdown_, i can do it be declaring a text reference using `(ref:[---])@[---]` and refer to that text using _paste0_ as instructed it in [_bookdown_'s instructions](https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html). This doesn't work with _quarto_. It isn't even documented in _quarto_'s instructions. How can i cite something inside a caption whose table is generated using _kable_?

**Comment by jjallaire** [comment]
There are no special hacks required, you can use markdown citations anywhere. So the following both work:

````markdown
```{r}
library(knitr)
kable(head(pressure), caption="Pressure [@spacean2000]")
```

```{r}
#| tbl-cap: Pressure [@spacean2000]
library(knitr)
kable(head(pressure))
```
````

Docs on citations here: https://quarto.org/docs/authoring/footnotes-and-citations.html

Docs on table captions w/ computations here (note that `tbl-cap` is the technique demonstrated as that works w/ all table output -- unlike kable some table generation packages don't actually have a caption option!): https://quarto.org/docs/authoring/tables.html#computations

**Reply by piiskop** [reply]
Your examples are legit. However, once i let read in _kableExtra_, even your examples don't work anymore. However, using references in captions using _kableExtra_ works in _bookdown_. i use _kableExtra_ for _kable\_styling_.

**Comment by jjallaire** [comment]
That's likely because kableExtra is emitting raw HTML which of course the markdown doesn't resolve in (thus the `ref` backdoor you mentioned in bookdown). We indeed do not currently have an equivalent back door. We will be investing considerably more effort into tables this fall which will probably lead to more of the kableExtra features being natively available.

**Comment by Minh-AnhHuynh** [comment]
Unless I'm missing something, I would like to add that the Quarto option can appropriately format citations (`@reference` stays raw) but can't do short captions, while kable can do caption.short but fails to cite... there's no win there. I'll have to hard code my citation in the meantime. This is for tables specifically from an R computation chunk. For regular tables since to my knowledge, quarto doesn't support short table captions, I use raw latex in the meantime.
